tree-walk.c

```diff
@@ -547,13 +564,26 @@ int get_tree_entry(const struct object_id *tree_oid, const char *name, struct ob
 		retval = -1;
 	} else {
 		struct tree_desc t;
-		init_tree_desc(&t, tree, size);
-		retval = find_tree_entry(&t, name, oid, mode);
+		if (gently) {
+			if (init_tree_desc_gently(&t, tree, size)) {
+				free(tree);
+				return -1;
+			}
+		} else {
+			init_tree_desc(&t, tree, size);
+		}
+		retval = find_tree_entry_gently(&t, name, oid, mode, gently);
 	}
 	free(tree);
 	return retval;
 }
 
+int get_tree_entry(const struct object_id *tree_oid, const char *name,
```

Comment:
I guess this function is used in too many places (23, if I counted correctly) to change the signature: You would have to add tons of `, 0`, making the patch harder to read. So I'd be okay with renaming `get_tree_entry()` (but I would suggest describing the reason in the commit message).